ui/app/src/components/PathNavigatorTree/PathNavigatorTree.tsx

```diff
@@ -419,6 +425,46 @@ export default function PathNavigatorTree(props: PathNavigatorTreeProps) {
       })
     );
   };
+
+  const onPreview = (item: DetailedItem) => {
+    if (item.systemType === 'component' || item.systemType === 'taxonomy') {
+      const src = `${legacyFormSrc}site=${site}&path=${item.path}&type=form&readonly=true`;
+      dispatch(showEditDialog({ src }));
+    } else if (item.mimeType.startsWith('image/')) {
+      dispatch(
+        showPreviewDialog({
+          type: 'image',
+          title: item.label,
+          url: item.path
+        })
+      );
+    } else {
+      fetchContentXML(site, item.path).subscribe((content) => {
```

Comment:
This here isn’t great either. 
Also, there’s no visibility of system status while loading the content.